core/src/main/java/com/linecorp/armeria/client/AbstractRuleBuilder.java

```diff
@@ -174,6 +175,18 @@ public AbstractRuleBuilder onException() {
         return onException((unused1, unused2) -> true);
     }
 
+    /**
+     * Adds {@link TimeoutException}.
+     */
+    public AbstractRuleBuilder onResponseTimeout() {
+        return onException((ctx, ex) -> {
```

Comment:
If `ctx.isTimedOut()` is true, we don't need type checking. 
```java
if (ctx.isTimedOut()) {
    return true;
}

return ex instanceof UnprocessedRequestException && 
       ex.getCause() instanceof TimeoutException;
}
```

Reply:
✅ updated!